Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -246,7 +246,7 @@ void Document::layout()
     m_layout_root->layout();
     m_layout_root->set_needs_display();
 
-    frame()->page().client().page_did_layout();
+    frame()->page().client().page_did_layout(this);
```

Comment:
Instead of this, let's just only send this notification to the `PageClient` when the *main frame* is laid out.
It doesn't really need to know about subframe layouts anyway since that does not affect widget-level layout.

```suggestion
    if (frame()->is_main_frame())
        frame()->page().client().page_did_layout();
```